src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs

```diff
@@ -323,5 +323,20 @@ public static class WellKnownMemberNames
         /// The required name for the <c>DisposeAsync</c> method used in an await using statement.
         /// </summary>
         public const string DisposeAsyncMethodName = "DisposeAsync";
+
+        /// <summary>
+        /// The required name for the <c>Count</c> property used in a pattern-based Index or Range indexer.
+        /// </summary>
+        public const string CountPropertyName = "Count";
```

Comment:
>public const string CountPropertyName = "Count"; [](start = 8, length = 48)

Why would we want to expose these constants publicly?

Reply:
Great question! I don't know, but it seems like all of these are public. Why should these members be different?

Reply:
>  Why should these members be different?

Why should we have these members in **this** type?

---
In reply to: [274721942](https://github.com/dotnet/roslyn/pull/34897#discussion_r274721942) [](ancestors = 274721942)

Reply:
It's where we store the names for other patterns, like foreach or async dispose. I could put them elsewhere, this just seemed like the natural place.